ignite/src/test/java/org/hibernate/ogm/datastore/ignite/test/cfg/LongIndexNameTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.ignite.test.cfg;
+
+import org.hibernate.HibernateException;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.Test;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import java.util.HashSet;
+import java.util.Set;
+
+import static junit.framework.TestCase.fail;
+
+/**
+ * Entities with long names that results in an index with greater than 255 chars is invalid in Ignite.
+ * This ensures we raise an error and provide useful information to the user
+ */
+public class LongIndexNameTest extends OgmTestCase {
+
+	@Test(expected = HibernateException.class)
+	public void testLongEntityIndexName() throws Exception {
+		fail( "The length of the registered entity's cache name should've failed this already" );
```

Comment:
This test is not good enough, The failure might be caused by something else.
How do you know that it failed because of the longer index? Actually, I don't think that's the cause of the failure. I think It's because you use `@Entity( name="..." )` instead of `@Table(name-=""`). But I might be wrong because I didn't run the test.

Reply:
I think `JPAAPIWrappingTest#testUndefinedPU()` might look similar to what you need.
You might want to have the following assertions (using the JUnit rule):
```
thrown.expect( HibernateException.class );
thrown.expectMessage( ... );
```

THere are multiple example of tests in hibernate ogm core using `ExpectedException` junit rule in the sources of hibernate-ogm core project.

Reply:
By the way, sometimes the JUnit rule is not easy to use. he try catch block is still a valid option